### PR TITLE
feat: user assigned managed identity & OIDC

### PR DESCRIPTION
**Pull request by matt-FFFFFF**
# Overview/summary

Adds support for creating a user assigned managed ideneity in the supplied subscription.

This is useful in two scenarios:

- granting access from your devops cicd pipeline runner virtual machine
- using with cloud based cicd runners and OpenID Connect, e.g. GH actions or Terraform cloud

This PR also repurposes the previously released networkwatcherrg submodule to create any number of given resource groups in the new subscriptions.

It is then possible to grant the umi access to these resource groups using the umi_role_assignments variable.

Have raised AzAPI minimum to 1.4.0

## Testing evidence

Please provide testing evidence to show that your Pull Request works/fixes as described and documented above.

## As part of this pull request I have

- [x] Checked for duplicate [Pull Requests](https://github.com/Azure/terraform-azurerm-lz-vending/pulls)
- [x] Associated it with relevant [issues](https://github.com/Azure/terraform-azurerm-lz-vending/issues), for tracking and closure.
- [x] Run and `make fmt` & `make docs` to format your code and update documentation.
- [x] Created unit and deployment tests and provided evidence.
- [x] Updated relevant and associated documentation.

### CODE REVIEW

**Review by jtracey93 — DISMISSED** [review]

File: modules/resourceprovider/header.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# ALZ landing zone resource provider submodule
```
Comment (jtracey93) [inline modules/resourceprovider/header.md:1]: ALZ?

File: modules/resourceprovider/README.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
+<!-- BEGIN_TF_DOCS -->
+# ALZ landing zone resource provider submodule
```
Comment (jtracey93) [inline modules/resourceprovider/README.md:2]: ALZ?

**Review by jtracey93 — CHANGES_REQUESTED** [review]

File: modules/resourceprovider/header.md (hunk 1 above)
Comment (matt-FFFFFF) [reply modules/resourceprovider/header.md:1]: fair!

**Review by jtracey93 — APPROVED** [review]
LGTM

**Pull request merged**